### PR TITLE
dgen: init at 1.33

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/emulators/dgen-sdl/default.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+, libarchive
+, doxygen
+, SDL
+}:
+
+let
+  pname = "dgen-sdl";
+  version = "1.33";
+in stdenv.mkDerivation {
+  inherit pname version;
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dgen-sdl/default.nix:12]:
```suggestion
in stdenv.mkDerivation  rec {
  pname = "dgen-sdl";
  version = "1.33";
```

Please stop doing this.

File: pkgs/misc/emulators/dgen-sdl/default.nix [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - Z80 debugger
+      - hqx and scale2x upscaling filters
+      - VGM dumping
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ AndersonTorres ];
+    platforms = with platforms; unix;
+  };
+}
+# TODO: handle configure options
```
Comment (SuperSandro2000) [inline pkgs/misc/emulators/dgen-sdl/default.nix:55]: What happens with the TODO?

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 106359` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>dgen-sdl</li>
  </ul>
</details>

File: pkgs/misc/emulators/dgen-sdl/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/misc/emulators/dgen-sdl/default.nix:55]: `dgen` has some interesting options, as building debuggers and emulating some extra processors.

I don know what is the better approach here: using a `config` parameter (with many `enable` subparameters) or directly listing possibilites as `enableDebugger, enableCoProcessor` etc.

Anyway, at least it runs Sonic!

Comment (SuperSandro2000) [reply pkgs/misc/emulators/dgen-sdl/default.nix]: > directly listing possibilites as enableDebugger, enableCoProcessor etc.

That is what I saw so far for packages. A config parameter is only used for modules.

File: pkgs/misc/emulators/dgen-sdl/default.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/misc/emulators/dgen-sdl/default.nix:12]: Why?

**Pull request merged**